tests/foreman/cli/test_contentview.py

```diff
@@ -5136,7 +5139,49 @@ class ContentViewFileRepoTestCase(CLITestCase):
     """Specific tests for Content Views with File Repositories containing
     arbitrary files
     """
-    @stubbed()
+    @classmethod
+    def setUpClass(cls):
+        """Create a product and an org which can be re-used in tests."""
+        super(ContentViewFileRepoTestCase, cls).setUpClass()
+        cls.org = make_org()
+        cls.product = make_product({'organization-id': cls.org['id']})
+
+    def _make_file_repository_upload_contents(self, options=None):
+        """Makes a new File repository, Upload File/Multiple Files
+        and asserts its success """
+        if options is None:
+            options = {}
+
+        options_list = (u'product-id', u'content-type', u'url', u'name')
+        if None in [options.get(keys) for keys in options_list]:
```

Comment:
I'd write `key` since it's singular. You can also write it as `if any(options.get(key) is None for key in options_list):` but the difference is probably negligible.

Reply:
I really do not like this form if some one is missing some option this will lead to an unpredictable behavior, think better to raise exception. But if options is None, initiate the options with the default one eg:
```python 
if options is None:
    options = {
          'product-id' : self.product['id']
          ...
    }
```
what do you think ?